src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String CREATE_QUERY = "INSERT INTO manufacturers (name, country) "
+            + "VALUES (?, ?)";
+    private static final String GET_QUERY = "SELECT * FROM manufacturers WHERE id = ?";
+    private static final String GET_ALL_QUERY = "SELECT * FROM manufacturers";
+    private static final String UPDATE_QUERY = "UPDATE manufacturers SET name = ?, "
+            + "country = ? WHERE id = ?";
+    private static final String DELETE_QUERY = "UPDATE manufacturers SET is_deleted = 1 "
+            + "WHERE id = ?";
+
+    private final Connection connection;
+
+    public ManufacturerDaoImpl(Connection connection) {
+        this.connection = connection;
+    }
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (PreparedStatement statement = connection.prepareStatement(CREATE_QUERY,
+                PreparedStatement.RETURN_GENERATED_KEYS)) {

```

Comment:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
        PreparedStatement statement = connection.prepareStatement(CREATE_QUERY,
                PreparedStatement.RETURN_GENERATED_KEYS)) {

```

for other method same

Reply:
Connection needs to be closed